src/network/networkd-json.c

```diff
@@ -1193,6 +1193,28 @@ static int dhcp_client_private_options_append_json(Link *link, JsonVariant **v)
         return json_variant_set_field_non_null(v, "PrivateOptions", array);
 }
 
+static int dhcp_client_client_id_append_json(Link *link, JsonVariant **v) {
+       const sd_dhcp_client_id *client_id;
+       _cleanup_free_ char *id = NULL;
+        int r;
+
+        assert(link);
+        assert(v);
+
+        if (!link->dhcp_client || !link->dhcp_lease)
+                return 0;
+
+        r = sd_dhcp_lease_get_client_id(link->dhcp_lease, &client_id);
+        if (r < 0)
+                return r;
+
+        r = sd_dhcp_client_id_to_string(client_id, &id);
```

Comment:
hmm, shouldn't this also be formatted as json object? seems like cheating to output this as string

Reply:
Hehe updated :)